lab_1_classify_by_unigrams/main.py

```diff
@@ -84,6 +156,32 @@ def compare_profiles(
     In case of corrupt input arguments or lack of keys 'name' and
     'freq' in arguments, None is returned
     """
+    if (profiles_bad_input(unknown_profile) is None or
+            profiles_bad_input(profile_to_compare) is None):
+        return None
+    unk_profile = copy.deepcopy(unknown_profile)
+    from_profile1 = unk_profile["freq"]
+    from_profile2 = profile_to_compare["freq"]
+    keys_for_1 = set(from_profile1.keys())
+    keys_for_2 = set(from_profile2.keys())
+    if keys_for_1.intersection(keys_for_2) == {}:
+        return None
+    letters_need1 = keys_for_1.difference(keys_for_2)
+    letters_need2 = keys_for_2.difference(keys_for_1)
+    for i in letters_need1:
+        from_profile2.setdefault(i, 0.0)
+    for x in letters_need2:
+        from_profile1.setdefault(x, 0.0)
+    sorted1 = dict(sorted(from_profile1.items()))
+    sorted2 = {}
+    for key in sorted1:
+        sorted2[key] = from_profile2.get(key, from_profile1[key])
+    list1 = list(sorted1.values())
+    list2 = list(sorted2.values())
+    conclusion = calculate_mse(list2, list1)
+    if not isinstance(conclusion, float):
+        return None
+    return round(conclusion, 4)
```

Comment:
why are rounding mse_function output up to 4 when this function already returns rounded number up to 4?

Reply:
done